python/nvfuser/benchmark_utils.py

```diff
@@ -22,67 +46,109 @@ def cleanup(self):
         pass
 
 
-class TorchProfileTimer(Timer):
+def cupti_call_safe(func, *args):
+    """Wrapper for CUPTI calls. Failing CUPTI calls will exit the program."""
+    try:
+        return func(*args)
+    except Exception as e:
+        print(f"CUPTI call {func.__name__} failed: {e}")
+        sys.exit(1)
+
+
+class CuptiProfiler:
+    # List of activities to be recorded by CUPTI.
+    activity_kinds = [
+        cupti.ActivityKind.CONCURRENT_KERNEL,
+    ]
+    _subscriber_handle = None
+
+    @staticmethod
+    def enable_cupti_activities():
+        for activity_kind in CuptiProfiler.activity_kinds:
+            cupti_call_safe(cupti.activity_enable, activity_kind)
+
+    @staticmethod
+    def disable_cupti_activities():
+        for activity_kind in CuptiProfiler.activity_kinds:
+            cupti_call_safe(cupti.activity_disable, activity_kind)
+
+    @staticmethod
+    def func_buffer_requested():
+        buffer_size = 8 * 1024 * 1024
+        max_num_records = 0
+        return buffer_size, max_num_records
```

Comment:
I'm not familiar with cupti API, would be help to have a comment about where the magic numbers are coming from.